pipe/pipe_param.py

```diff
@@ -16,7 +16,8 @@ class PipeParam:
     """ PipeParam keeps all parameters relevant for the extraction, with
     default values given for most.
     """
-    def __init__(self, name, visit, version=None, outdir=None, sa_range=None):
+    def __init__(self, name, visit, version=None, outdir=None, sa_range=None,
+                 datapath=None, calibpath=None):

```

Comment:
I've added a `datapath` and `calibpath` argument here so that we can direct the paths to a predefined path other than one constructed from `name` and `visit`, which is useful for testing. Doesn't break backwards compatibility.